source/dofs/dof_handler.cc

```diff
@@ -1096,36 +1096,51 @@ template <int dim, int spacedim>
 types::global_dof_index
 DoFHandler<dim, spacedim>::n_boundary_dofs() const
 {
-  std::set<int> boundary_dofs;
-
-  const unsigned int                   dofs_per_face = get_fe().dofs_per_face;
-  std::vector<types::global_dof_index> dofs_on_face(dofs_per_face);
-
-  // loop over all faces of all cells
-  // and see whether they are at a
-  // boundary. note (i) that we visit
-  // interior faces twice (which we
-  // don't care about) but exterior
-  // faces only once as is
-  // appropriate, and (ii) that we
-  // need not take special care of
-  // single lines (using
-  // @p{cell->has_boundary_lines}),
-  // since we do not support
-  // boundaries of dimension dim-2,
-  // and so every boundary line is
+  const FiniteElement<dim, spacedim> &fe            = get_fe();
+  const unsigned int                  dofs_per_face = fe.n_dofs_per_face();
+  std::vector<dealii::types::global_dof_index> dofs_on_face(dofs_per_face);
+
+  const IndexSet owned_dofs = locally_owned_dofs();
```

Comment:
That function returns a reference. Would you mind only storing a reference instead of copying it?